### ISSUE TITLE
Taxonomy Rosetta Stone

### ISSUE THREAD

**Issue by CloVasco** [benetech/ServiceNet#319]
Translate partners’ taxonomy into their own once we can develop the “taxonomy Rosetta stone”

**Comment by NeilMBenetech** [comment]
It's possible the partners themselves might want to maintain the "crosswalk" from their own taxonomy systems' to their various partners'. It would be a big task but even when using the standard AIRS taxonomy, each agency tends to highly customize which terms they use - it is common for them to de-activate up to 90% of the 10,000 taxonomy terms, for example, in ways that are very specific to their geographic and topical focuses. For that reason we might not be able to come up with our own "one size fits all" taxonomy rosetta stone.

**Comment by johnhbenetech** [comment]
@NeilMBenetech We also had some questions/concerns about the licensing requirements on the AIRS side. i.e. What are we as a third party consumer allowed to do/see/display with regards to specific terms. I'm assuming you'd have some clarity on that.

**Comment by NeilMBenetech** [comment]
@johnhbenetech We'd likely need to get our own license as a vendor, from 2-1-1 Los Angeles (the owners). They differentiate between licenses for software vendors (can't currently find link or info on their website) and licenses/subscriptions for "consumers" of the taxonomy. https://211taxonomy.org/ . The vendor licenses used to be $2,500 per year. When ready we can enquire with them.

Relatedly there seems to be a rift emerging between AIRS and 211 LA regarding the taxonomy itself. This was publicly announced back in April 2019, I have a little bit of the backstory but not much more : "After careful consideration and following the expiry of our MOU, the AIRS Board recently informed 211 LA County that the AIRS logo and branding will no longer be affiliated with the 211 LA County Taxonomy of Human Services. AIRS will continue to support the day-to-day work of the AIRS Taxonomy Committee and the Taxonomy's Networker group. We will also continue to provide training for the Taxonomy (for example, within the AIRS I&R Training Manual and at the Conference). Over the next few months, the AIRS Board will be assessing the situation and will keep members informed of any relevant developments."

**Comment by greggish** [comment]
i agree with neil that there may not be a 'one size fits all' taxonomy, but if by 'rosetta stone' you mean 'translation interface that can make it easier to create and maintain taxonomy crosswalks,' that would be extremely valuable. 

[The Joined-up Data Standards project prototyped a code translation tool](http://www.joinedupdata.org/en/spreadsheet/), i think, that may provide some of this initial functionality.

**Comment by NeilMBenetech** [comment]
Hi @johnhbenetech , let's explore internally whatever approach we want to take with 211 LA about a mutually beneficial relationship between them and Benetech for Service Net to have a license to the taxonomy.

**Comment by NeilMBenetech** [comment]
@johnhbenetech I propose we move forward with this now. 211 LA will grant us a license, we just haven't had the call with them and Betsy yet so the terms are not yet settled. But it will be an important component of matching at the Service level, which itself is an important part of improving the "Workflow" of users.

For our first iteration I propose that we don't need UI for this, but rather the mechanism itself. Any given partner's categorization system should be able to be linked on a many-to-many basis to any/all other partners' categorization systems. The links should be able to be established at any level of hierarchy of any of the systems. 

For testing purposes, I also propose that you provide me with an Excel spreadsheet template containing the PK, name, and parent information of each category for every partner, and some structure in which I can do the matching to any other partner's categorization system. I'll do some initial matching, manually, and return the template to you for upload into Service Net.

I'll create a related task now about how this Taxonomy Rosetta Stone will be used for matching purposes.

**Comment by johnhbenetech** [comment]
For the first subtask, let's create a SQL query that will group partner taxonomical terms by % related based on matching organizations.

**Comment by johnhbenetech** [comment]
@NeilMBenetech We are almost ready to send you some spreadsheets for you to assist in the manual matching process.

While you do that the plan is to build out the linkage database table that will store the results of your analysis.

One question for now - do you think we should create an admin page allowing you/someone from benetech to be able to add/remove these matches on the fly?

**Comment by NeilMBenetech** [comment]
Hey @johnhbenetech a UI to manage the matches would be wonderful - I left it out of scope for this task just to keep things more manageable for now, while we iterate. If you do think we could squeeze in UI, it would be important to respect any hierarchy in each categorization scheme, as we know the AIRS Taxonomy has for example. 

For that matter, I erroneously omitted the effect of hierarchy on this overall task. Here is that refinement: When you link together a taxonomy term/category from any given system to any other system, then all "child" terms/categories should be treated as being linked as well. In more concrete terms, if I indicate a particular taxonomy term at a depth of (arbitrarily for this example) level 3 of the taxonomy is linked to some other category term in another partner's system, then all taxonomy terms below that term (level 4 through down to level 6 in this example, the depth can vary from branch to branch) should also be considered to be linked. In the AIRS taxonomy, the taxonomy codes themselves reflect the hierarchy so you could put a LIKE clause in your SQL to cause this to work properly.....such as : "....AND TaxonomyCode LIKE 'BD_2000%'"

**Comment by johnhbenetech** [comment]
@NeilMBenetech so as I see it we still need a few inputs from your side, let me know if I missed something:

1. For non AIRS, we need recommendations from you on how we should link SMC, LAAC, and healthleads service types (both between themselves and to AIRS).

2. For AIRS, do you think we should just start by only using a specific level? Like for instance just starting with level 3? In that case any other AIRS with the same level 3 term would get a match, and you would only have to related SMC, LAAC, and healthleads to level 3 terms?

**Comment by NeilMBenetech** [comment]
@johnhbenetech yes indeed, I'll do at least some preliminary mapping from the spreadsheet you sent me amongst all the partners. 

And for the AIRS taxonomy I will certainlly be aggregating up to level 2 or 3. Each branch has some subtelties about how deep you need to go to get meaningfully targeted so it'll be different for each one. 

![Capture](https://user-images.githubusercontent.com/52297301/68636505-8a580180-04b0-11ea-8796-b46321034e0d.PNG)

**Comment by bognasoldev89** [comment]
At this moment Taxonomy group is available in the admin UI that allows creating taxonomies relations in DB.
I created 2 exemplary groups in staging environment (and I was able to edit them) but it only works for DB at this moment - user is not able to filter taxonomy group from the homepage.

![319 1](https://user-images.githubusercontent.com/51324797/72361727-44a6f780-36f2-11ea-9f42-ac1a186b8849.png)

![319 2](https://user-images.githubusercontent.com/51324797/72361729-44a6f780-36f2-11ea-8a9f-6e44a3fc53b1.png)

@johnhbenetech @NeilMBenetech how to proceed?

**Comment by bognasoldev89** [comment]
@johnhbenetech as we spoke during a check-up, I'm moving this ticket back to Up next column

**Comment by kkaczmarczyk** [comment]
@johnhbenetech @jkondrat is there a clear idea of how we should go about this issue yet?

**Comment by greggish** [comment]
Chiming back in here: in this upgrade cycle for HSDS, one of the big prospective changes that we're discussing might relate to your work on this functionality. See openreferral/specification#214 and openreferral/specification#181 

It would be great to hear your perspective on working among multiple taxonomies as we consider these options.

**Comment by johnhbenetech** [comment]
@NeilMBenetech Is there an update on this? What's your plan for populating this table?

**Comment by NeilMBenetech** [comment]
Sorry @johnhbenetech for the delay, I will prioritize.

**Comment by greggish** [comment]
The Learned Hands project (working to crosswalk national taxonomies of legal issues) may be onto something applicable here: https://github.com/learned-hands/mapping_aid/blob/master/README.md
